### PR TITLE
Add tests for `get_node` autocompletion

### PR DESCRIPTION
**Pull request by HolonProduction**
Requires #86961

### CODE REVIEW

**Comment by Mickeon** [comment]
A *whooping* 100 files added. Are they all truly necessary for the tests?

**Comment by HolonProduction** [comment]
> A _whooping_ 100 files added. Are they all truly necessary for the tests?

Yeah the file count kinda exploded :sweat_smile:. But every idea I have to reduce it seems worse.

- Each test needs at least 2 Files. That makes 50 test, but some of them are duplicated in different call situations (yes this is needed, see for example [#72328](https://github.com/godotengine/godot/issues/72328) and there also was another one that i cant find rn :disappointed:). So we are at ~30 test, and I plan to do more, not reduce them. For something as complex as this, reducing tests doesn't seem like a good idea.
- I often have test which require the same output for multiple files. But specifying script paths inside of the cfg files, would make it harder to find the right cfg. And I don't wan't to have multiple scripts inside of the same file, since it would introduce yet another "magic" character to separate them.
- We could reduce each test into one file, but this would either require a sort of preface for the GDScript and parsing that would be annoying. Or we could embed the GDScript inside the cfg, but that would make it impossible to test something manually by opening the script in the editor.

If you have any other ideas to reduce it, I'm all in for it, that being said many test files aren't that big of a problem to justify any of the tradeoffs that I listed IMO.

File: modules/gdscript/tests/test_completion.h
Comment (AThousandShips) [inline modules/gdscript/tests/test_completion.h:35]: I'd put these among the rest, none of the tests are run without tools anyway so no need for includes

**Comment by HolonProduction** [comment]
Waiting for #87759 in order to fix the CI.

**Comment by HolonProduction** [comment]
Ready for review

**Review by vnen — APPROVED** [review]
On a cursory look it is good to me. Thanks for working on this!

**Comment by akien-mga** [comment]
Thanks!

**Comment by HolonProduction** [comment]
Oh shit, I just noticed I forgot to squash the commits and now you merged it :sweat_smile:. I guess it can't be changed now?

**Comment by akien-mga** [comment]
Yeah I missed that. It can't be changed once merged.

**Pull request merged**
